### PR TITLE
auth: Stop sending api key in query string

### PR DESCRIPTION
**Pull request by nghiant2710**
The SDK should never send the api key as query string and only via http headers

Connects-to: https://github.com/balena-io/balena-sdk/pull/780

Change-type: major
Signed-off-by: Trong Nghia Nguyen <nghiant2710@gmail.com>

### CODE REVIEW

**Review by afitzek — APPROVED** [review]
lgtm

File: balena/base_request.py [hunk 1]
```diff
@@ -137,11 +137,11 @@ def __request(self, url, method, params, endpoint, headers=None,
...
         if auth:
-            if not api_key:
-                self.__set_authorization(headers)
-            params = self._format_params(params, api_key=api_key, raw_query=raw_query)
+            # api_key will be None if Auth token is found so we can just pass api_key here.
+            self.__set_authorization(headers, api_key)
+            params = self._format_params(params, raw_query=raw_query)
         else:
-            params = self._format_params(params, api_key=None, raw_query=raw_query)
+            params = self._format_params(params, raw_query=raw_query)
```
Comment (thgreasi) [inline balena/base_request.py]: Let's move this line outside of the `if-else` and drop the `else`.
```suggestion
            params = self._format_params(params, raw_query=raw_query)
```

**Review by thgreasi — APPROVED** [review]

File: balena/base_request.py (hunk 1 above)
Comment (nghiant2710) [reply balena/base_request.py]: Updated!

**Comment by thgreasi** [comment]
Already approved :+1:

**Pull request merged**